src/multiParamManager.js

```diff
@@ -20,16 +21,50 @@ class MultiParamManager {
     this.clickCallBack = clickCallBack
     this.inputs = inputs
     this.title = title
+    this.basicInputField
+    this.multiFields
   }
 
   switchMethodViewOn () {
     this.contractActionsContainerSingle.style.display = 'none'
     this.contractActionsContainerMulti.style.display = 'flex'
+    // fill in the inputs
+    this.makeMultiVal()
   }
 
   switchMethodViewOff () {
     this.contractActionsContainerSingle.style.display = 'flex'
     this.contractActionsContainerMulti.style.display = 'none'
+    this.basicInputField.value = this.getMultiValsString()
+  }
+  getMultiValsString () {
+    var valArray = this.multiFields.querySelectorAll('input')
+    var ret = ''
+    for (var k = 0; k < valArray.length; k++) {
+      var el = valArray[k]
+      if (ret !== '') ret += ','
+      ret += el.value
+    }
+    return ret
+  }
+
+  emptyInputs () {
+    var valArray = this.multiFields.querySelectorAll('input')
+    for (var k = 0; k < valArray.length; k++) {
+      valArray[k].value = ''
+    }
+    this.basicInputField.value = ''
+  }
+
+  makeMultiVal () {
+    var inputString = this.basicInputField.value
```

Comment:
I think we might need to wrap with double quote here
```
 params = params.replace(/(^|,\s+|,)(\d+)(\s+,|,|$)/g, '$1"$2"$3') // replace non quoted number by quoted number
        params = params.replace(/(^|,\s+|,)(0[xX][0-9a-fA-F]+)(\s+,|,|$)/g, '$1"$2"$3') // replace non quoted hex string by quoted hex string
```
cause the JSON.parse will break if some values are not wrapped ...